pkg/monitor/dissect.go

```diff
@@ -220,52 +214,52 @@ func GetConnectionSummary(data []byte, opts *decodeOpts) string {
 
 // Dissect parses and prints the provided data if dissect is set to true,
 // otherwise the data is printed as HEX output
-func Dissect(dissect bool, data []byte) {
-	if dissect {
-		dissectLock.Lock()
-		defer dissectLock.Unlock()
-
-		initParser()
+func Dissect(buf *bufio.Writer, dissect bool, data []byte) {
```

Comment:
Commit `cleanup: early return in Dissect method`: I also prefer early exists, but just to make sure this is only a refactor right? I don't see functional changes wrt the previous if-else.

We can keep the commit, would just rename it `monitor: early return in Dissect method` or `monitor:refactor: unwind if-else in Dissect method`.